### PR TITLE
DT-645 Duplicate offenders have NOMS number moved

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/delius/integration/secure/CustodyUpdateNomsNumberAPITest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles("dev-seed")
+@DirtiesContext
+@Disabled("Disabled until CI memory issues are addressed")
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/integration/secure/CustodyUpdateNomsNumberAPITest.java:39]: Sadly these are still disabled until the H2 out of memory issue is fixed (on PiCT team backlog)

File: src/test/java/uk/gov/justice/digital/delius/service/OffenderIdentifierServiceTest.java [hunk 2]
```diff
@@ -56,93 +62,149 @@ void willNotUpdateNOMSNumber() {
     class FeatureSwitchedOn {
         @BeforeEach
         void setUp() {
-            service = new OffenderIdentifierService(true, new OffenderTransformer(new ContactTransformer()), offenderRepository, spgNotificationService);
+            service = new OffenderIdentifierService(true, new OffenderTransformer(new ContactTransformer()), offenderRepository, spgNotificationService, referenceDataService);
+            when(offenderRepository.findByNomsNumber(any())).thenReturn(Optional.empty());
         }
 
-        @Test
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/service/OffenderIdentifierServiceTest.java:62]: Hard to see what has changed in mixed view, side by side might be better

File: src/test/java/uk/gov/justice/digital/delius/integration/secure/CustodyUpdateNomsNumberAPITest.java (hunk 1 above)
Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/digital/delius/integration/secure/CustodyUpdateNomsNumberAPITest.java:39]: Is it worth sticking a TODO on there pointing at the PiCT JIRA ticket?

File: src/test/java/uk/gov/justice/digital/delius/service/OffenderIdentifierServiceTest.java [hunk 3]
```diff
@@ -56,93 +62,149 @@ void willNotUpdateNOMSNumber() {
...
+            void willUpdateOffender() {
+                assertThat(iDs.getNomsNumber()).isEqualTo("G5555TT");
             }
 
+            @Test
+            void willUpdateDuplicateOffender() {
+                verify(spgNotificationService, times(2)).notifyUpdateOfOffender(offenderCaptor.capture());
+
+                // can only test this via spg call since this is a side affect
+                final Offender duplicateOffender = offenderCaptor.getAllValues().get(0);
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/delius/service/OffenderIdentifierServiceTest.java:195]: Could use a var here?

File: src/test/java/uk/gov/justice/digital/delius/integration/secure/CustodyUpdateNomsNumberAPITest.java (hunk 1 above)
Comment (andymarke) [reply src/test/java/uk/gov/justice/digital/delius/integration/secure/CustodyUpdateNomsNumberAPITest.java:39]: Good idea

File: src/test/java/uk/gov/justice/digital/delius/service/OffenderIdentifierServiceTest.java (hunk 3 above)
Comment (andymarke) [reply src/test/java/uk/gov/justice/digital/delius/service/OffenderIdentifierServiceTest.java:195]: yep, I'll change in the next PR

**Pull request merged**